coins/monero/src/rpc.rs

```diff
@@ -513,16 +513,27 @@ impl<R: RpcConnection> Rpc<R> {
   /// Get the currently estimated fee from the node. This may be manipulated to unsafe levels and
   /// MUST be sanity checked.
   // TODO: Take a sanity check argument
-  pub async fn get_fee(&self) -> Result<Fee, RpcError> {
+  pub async fn get_fee(&self, priority: u32) -> Result<Fee, RpcError> {
     #[allow(dead_code)]
     #[derive(Deserialize, Debug)]
     struct FeeResponse {
-      fee: u64,
+      fees: Vec<u64>,
       quantization_mask: u64,
     }
 
     let res: FeeResponse = self.json_rpc_call("get_fee_estimate", None).await?;
-    Ok(Fee { per_weight: res.fee, mask: res.quantization_mask })
+
+    let priority_idx =
+      if priority >= 4 { 3 } else { priority.saturating_sub(1).try_into().unwrap() };
+
+    debug_assert!(priority_idx < res.fees.len(), "Unexpected size of fee response");
```

Comment:
This allows a malicious node to crash any app with debug assertions. Please use an error here since this can't be stated as an invariant.

Reply:
Fair.